File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    distanceFromCenter = distance({x, y})
    cond do
      distanceFromCenter <= 1 -> 10
      distanceFromCenter <= 5 -> 5
      distanceFromCenter <= 10 -> 1
      true -> 0
    end
  end

  def distance({x, y}) do
    (:math.pow(Kernel.abs(x),2) + :math.pow(Kernel.abs(y), 2)) |> :math.sqrt
  end
end

File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"

#include <cmath>

namespace darts {
    int score(double x, double y) {
        double distance = sqrt(pow(x, 2) + pow(y, 2));
        if (distance <= 1) {
            return 10;
        }
        if (distance <= 5) {
            return 5;
        }
        if (distance <= 10) {
            return 1;
        }
        return 0;
    }
} // namespace darts